core/node/rules/can_add_event.go

```diff
@@ -634,6 +640,62 @@ func (params *aeParams) creatorIsMember() (bool, error) {
 	return true, nil
 }
 
+// getCreatorIsMember returns cached result of creator membership check.
+// This avoids duplicate IsMember() calls between check and chain auth functions.
+func (params *aeParams) getCreatorIsMember() (bool, error) {
+	if params.creatorMembershipCached {
+		return params.creatorIsMemberResult, params.creatorMembershipErr
+	}
+	params.creatorMembershipCached = true
+	params.creatorIsMemberResult, params.creatorMembershipErr = params.streamView.IsMember(
+		params.parsedEvent.Event.CreatorAddress,
+	)
+	return params.creatorIsMemberResult, params.creatorMembershipErr
+}
```

Comment:
please remove the caching, just use botOnwerChainAuthIfNotMember - please rename to what i suggested earlier.